geopyspark/geotrellis/layer.py

```diff
@@ -577,6 +581,32 @@ def collect_keys(self):
         else:
             return [temporal_projected_extent_decoder(key) for key in self.srdd.collectKeys()]
 
+    def merge_duplicate_keys(self, num_partitions=None):
```

Comment:
The name is a little loose and long, we're not merging keys, we're merging tiles with duplicate keys. We know they're duplicate because they have the same key. 

A little tough to come up with a name for it, so why not keep `merge`? Could there be a confusion that it means something else? (<- not rhetorical question, question question)

Other likely use for merge are two merge two layers together:

```py
layer1.merge() # merges duplicate keys

layer1 = catalog.query(...)
layer2 = catalog.query(...)
layer1.merge(layer2) # merges tiles from layer2 into tiles from layer1
```